genie-web/src/main/java/com/netflix/genie/web/controllers/UIController.java

```diff
@@ -74,6 +78,18 @@ public String getIndex(@NotNull final HttpServletResponse response, @Nullable fi
      */
     @RequestMapping(value = "/file/{id}/**", method = RequestMethod.GET)
     public String getFile(@PathVariable("id") final String id, final HttpServletRequest request) {
-        return "forward:/api/v3/jobs/" + id + "/" + ControllerUtils.getRemainingPath(request);
+        String encodedId;
+        try {
+            // When forwarding, the downstream ApplicationContext will perform URL decoding.
+            // If the job ID contains special characters (such as '+'), they will be interpreted as url-encoded and
+            // decoded, resulting in an invalid job ID that cannot be found.
+            // To prevent this, always perform URL encoding on the ID before forwarding.
+            encodedId = URLEncoder.encode(id, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            log.warn("Failed to encode id: {}", id, e);
+            encodedId = id;
```

Comment:
In practice this will never throw since "UTF-8" is constant and a valid encoding.